### PR TITLE
More proxy e2e tests - Stopping proxy service on source node

### PR DESCRIPTION
**Pull request by engelmi**
This PR adds the e2e test for stopping a proxy service on the source node (#221). In addition, it 
- simplifies copying systemd unit files to the container
- fixes the networking when running the e2e tests locally

### CODE REVIEW

File: tests/tests/tier0/proxy-service-start/test_proxy_service_start.py [hunk 1]
```diff
@@ -1,47 +1,49 @@
...
-def exec(ctrl: HirteContainer, nodes: Dict[str, HirteContainer]):
+def verify_proxy_start(ctrl: HirteControllerContainer, nodes: Dict[str, HirteNodeContainer]):
     foo = nodes[node_foo_name]
     bar = nodes[node_bar_name]
 
-    foo.create_file("/etc/systemd/system", "requesting.service", read_file("./systemd/requesting.service"))
-    foo.systemctl_daemon_reload()
-    bar.create_file("/etc/systemd/system", "simple.service", read_file("./systemd/simple.service"))
-    bar.systemctl_daemon_reload()
+    source_dir = os.path.join(".", "systemd")
```
Comment (Yarboa) [inline tests/tests/tier0/proxy-service-start/test_proxy_service_start.py]: This one is failing
Assume some think like this
```
source_dir = os.path.join(os.getcwd(),"proxy-service-start", "systemd")
```

or set environment var into tests with relevant path to pytest through tmt

File: tests/tests/tier0/proxy-service-start/test_proxy_service_start.py [hunk 2]
```diff
@@ -1,47 +1,49 @@
...
 
-    foo.wait_for_hirte_agent()
-    bar.wait_for_hirte_agent()
-    ctrl.wait_for_hirte()
+    foo.copy_systemd_service(requesting_service, source_dir, target_dir)
+    bar.copy_systemd_service(simple_service, source_dir, target_dir)
 
-    result, _ = ctrl.exec_run(f"hirtectl start {node_foo_name} requesting.service")
-    assert result == 0
+    result, output = ctrl.exec_run(f"hirtectl start {node_foo_name} {requesting_service}")
```
Comment (Yarboa) [inline tests/tests/tier0/proxy-service-start/test_proxy_service_start.py:29]: I think you should enable the service first 
systemctl enable inside the container before start 

I think you should enable it after the copy

**Review by Yarboa — CHANGES_REQUESTED** [review]

File: tests/tests/tier0/proxy-service-start/test_proxy_service_start.py (hunk 1 above)
Comment (engelmi) [reply tests/tests/tier0/proxy-service-start/test_proxy_service_start.py]: The tests were failing due to an error in `copy_systemd_service`s implementation where the service file was created at `/etc/systemd/system/<file_name>/<file_name>`. I removed the additional file_name in the path.

File: tests/tests/tier0/proxy-service-start/test_proxy_service_start.py (hunk 2 above)
Comment (engelmi) [reply tests/tests/tier0/proxy-service-start/test_proxy_service_start.py:29]: This shouldn't be necessary. `systemctl enable` will ensure the unit is started on the next boot, which is not needed in this test case.

**Review by mwperina — APPROVED** [review]
+1

**Comment by Yarboa** [comment]
@engelmi 
Is it still a draft? 
I assume Description should have some description
And which part of this #221 is covered?

**Review by Yarboa — APPROVED** [review]

**Comment by engelmi** [comment]
> @engelmi
> Is it still a draft?
> I assume Description should have some description
> And which part of this https://github.com/containers/hirte/issues/221 is covered?

@Yarboa Yes, it is supposed to be still in draft as I planned to collect all tests listed in #221 here. But as there are already some additions other than tests, I'd mark it as ready and update the description of this PR accordingly - so the PR doesn't get too big. 

Edit: Updated the description

**Pull request merged**